File: week07/32/241014.c
#include <stdio.h>
#include <stdlib.h>

int* getRow(int rowIndex, int* returnSize) {
    if (rowIndex == 0) {
        *returnSize = 1;
        int* row = (int*)malloc(*returnSize * sizeof(int));
        row[0] = 1;
        return row;
    } else if (rowIndex == 1) {
        *returnSize = 2;
        int* row = (int*)malloc(*returnSize * sizeof(int));
        row[0] = 1;
        row[1] = 1;
        return row;
    } else {
        int* previous = getRow(rowIndex - 1, returnSize);
        int currentSize = *returnSize + 1;
        int* current = (int*)malloc(currentSize * sizeof(int));
        current[0] = 1;

        for (int i = 1; i < *returnSize; i++) {
            current[i] = previous[i] + previous[i - 1];
        }
        current[currentSize - 1] = 1;

        free(previous);
        *returnSize = currentSize;
        return current;
    }
}

File: week07/32/241014.py
class Solution:
    def getRow(self, rowIndex):
        if rowIndex == 0:
            return [1]
        elif rowIndex == 1:
            return [1, 1]
        else:
            previous = self.getRow(rowIndex - 1)
            current = [1]
            for i in range(1, len(previous)):
                current.append(previous[i] + previous[i - 1])
            current.append(1)
            return current